src/fal/node_graph.py

```diff
@@ -148,13 +148,13 @@ def get_node(self, id: str) -> FalFlowNode | None:
         return self.node_lookup.get(id)
 
     def _is_script_node(self, node_name: str) -> bool:
-        return node_name.endswith(".py")
+        return _is_script(node_name)
 
     def _is_critical_node(self, node):
         successors = list(self.graph.successors(node))
         if node in successors:
             successors.remove(node)
-        has_script_pred = lambda node_name: node_name.endswith(".py")
+        has_script_pred = _is_script
         has_model_pred = lambda node_name: node_name.split(".")[0] == "model"
         if any(has_script_pred(i) for i in successors) and any(
```

Comment:
you can just change `has_script_pred` for `_is_script` and remove `has_script_pred` var.